engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -284,11 +271,13 @@ public void shouldActivateAndCompleteCompensationHandlerForIntermediateThrowEven
                 ProcessInstanceIntent.ELEMENT_COMPLETED,
                 PROCESS_ID));
 
-    // verify compensation subscription are removed from the state
-    final var compensations =
-        compensationSubscriptionState.findSubscriptionsByProcessInstanceKey(
-            TenantOwned.DEFAULT_TENANT_IDENTIFIER, processInstanceKey);
-    assertThat(compensations).isEmpty();
+    assertThat(
+            RecordingExporter.compensationSubscriptionRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limit(3))
+        .extracting(Record::getValueType, Record::getIntent)
+        .containsSubsequence(
```

Comment:
:wrench: We verify only a single event.

```suggestion
        .contains(
```